### PR TITLE
[DOCS] Migration Guide now includes example for Databases

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
### Docs Changes
- Added `tip` pointing users to example configurations in the `great_expectations` repo [here](https://github.com/great-expectations/great_expectations/tree/develop/tests/test_fixtures/configuration_for_testing_v2_v3_migration) . The Readme in that directory is also being updated as part of this PR. 
- Added tab for `Database` (should this be `Postgresql` instead?)
- Added `<detail>` tabs that explain parameters like `base_directory` and `query`, and point users to the example configurations. 
- Added list for V3-Datasource configurations for Database.  For the database, example, I felt it was better to list out the examples we have in the docs (1 step for user to take), rather than point them to the index page (2 steps for the user to take). 

### Fixtures
- Added `/postgresl` example to `tests/test_fixtures/configuration_for_testing_v2_v3_migration/`, as before this includes V2 and V3 style configurations that include Datasources and Checkpoints. 
- Included a Jupyter Notebook (`SetupTestDB.ipynb`) that allows users to load `Titanic.csv` that is used in all of our examples for the Migration Guide, and load it into a `postgresql` database that is running locally.  
-  Also, the `postgresql` database can be run using the Docker image that is part of the `great_expectations` repository, and is found at `great_expectations/assets/docker/postgresql` which the Notebook explains. 

### Tests
- Added 2 tests `migration_guide_postgresql_v2_api.py` and `migration_guide_postgresql_v3_api.py` that place the `postgresql` fixture under test. These test scripts use the refactor that was enabled in #4012 , and have been added to `test_script_runner.py`

### Associated JIRA Items
- GREAT-435 / GREAT-504 / GREAT-516

### Definition of Done
- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

### CODE REVIEW

File: docs/guides/miscellaneous/migration_guide.md [hunk 1]
```diff
@@ -155,15 +155,25 @@ Now you are ready to manually migrate Datasources and Checkpoints to be compatib
...
+
+:::
 
 <Tabs
   groupId="configurations-pandas-spark-postgres"
   defaultValue='pandas'
   values={[
   {label: 'Pandas', value:'pandas'},
   {label: 'Spark', value:'spark'},
+  {label: 'Database', value:'db'},
```
Comment (Shinnnyshinshin) [inline docs/guides/miscellaneous/migration_guide.md:174]: Currently the tab says `Database`. Should it be `Postgresql` instead?

File: tests/test_fixtures/configuration_for_testing_v2_v3_migration/README.md [hunk 2]
```diff
@@ -6,30 +6,33 @@ date: 20211022
 ## Overview
 
 - This folder contains configurations that were used to test the V2 to V3 migration guide found here : https://docs.greatexpectations.io/docs/guides/miscellaneous/migration_guide
-- It contains a complete-and-working V2 Configuration and a complete-and-working V3 Configuration that can be used to
+- It contains a complete-and-working V2 Configuration and a complete-and-working V3 Configuration that can be used to help with the migration process.
 
 ## So what's in the folder?
 
-- Test File : `Titanic.csv` that is found in the base directory
+- `/data` : This folder contains a test file, `Titanic.csv` that is used by the configurations in this directory.
```
Comment (cdkini) [inline tests/test_fixtures/configuration_for_testing_v2_v3_migration/README.md]: Should this be `data/`?

Comment (cdkini) [reply tests/test_fixtures/configuration_for_testing_v2_v3_migration/README.md]: Same with the other file paths below? `/` by itself is a filepath in UNIX system (points to binaries and some other underlying OS files), meaning that `/pandas` and `pandas/` point to different things.

File: docs/guides/miscellaneous/migration_guide.md (hunk 1 above)
Comment (cdkini) [reply docs/guides/miscellaneous/migration_guide.md:174]: I think Database is fine since you're using Postgres as a way to explain Database configuration

Comment (cdkini) [reply docs/guides/miscellaneous/migration_guide.md:174]: You can maybe point to the other DB guides at the end of the Postgres section if you want to make it general?

File: docs/guides/miscellaneous/migration_guide.md [hunk 3]
```diff
@@ -203,25 +217,71 @@ The V3-style Datasource has:
...
+Migrating Datasource configurations that contain connections to databases involve additional parameters like credentials that are specific to each configuration. The how-to-guides for Great Expectations contain numerous examples of V3 configurations that can be used for these various situations. 
+
+Please check out the following docs for examples of V3-style Datasource configurations that will suit your needs: 
+
+- [How to connect to a Athena database](/docs/guides/connecting_to_your_data/database/athena)
+- [How to connect to a BigQuery database](/docs/guides/connecting_to_your_data/database/bigquery)
+- [How to connect to a MySQL database](/docs/guides/connecting_to_your_data/database/mysql)
+- [How to connect to a Redshift database](/docs/guides/connecting_to_your_data/database/redshift)
+- [How to connect to a Snowflake database](/docs/guides/connecting_to_your_data/database/snowflake)
+- [How to connect to a SQLite database](/docs/guides/connecting_to_your_data/database/sqlite)
```
Comment (cdkini) [inline docs/guides/miscellaneous/migration_guide.md:281]: Just kidding looks like you covered my previous point 😅

**Review by cdkini — COMMENTED** [review]
Looks great! Happy to provide the sign-off but I'm not super well versed in V2 or our docs so I'll leave it open for the time being. Let me know if you need the approval!

File: tests/test_fixtures/configuration_for_testing_v2_v3_migration/README.md (hunk 2 above)
Comment (Shinnnyshinshin) [reply tests/test_fixtures/configuration_for_testing_v2_v3_migration/README.md]: Thank you for pointing this out @cdkini :) sorry for missing this

**Comment by Shinnnyshinshin** [comment]
really appreciate the review @cdkini. If you're ok with it, I'd like to merge this PR for now.  We will be re-visiting it soon in light of the feedback we received from PerPay.

**Review by cdkini — APPROVED** [review]
LGTM ❤️ nice work!

**Pull request merged**